src/test/java/com/mathworks/ci/MatlabBuilderTest.java

```diff
@@ -279,4 +282,134 @@ public void verifyCustomCommandInvokedForBatchMode() throws Exception {
+    public void verifyMatlabVersionOlderThanR17a() throws Exception {
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        RunTestsAutomaticallyOption runOption = new RunTestsAutomaticallyOption();
+        runOption.setTaCoberturaChkBx(true);
+        runOption.setTaJunitChkBx(true);
+        runOption.setTatapChkBx(true);
+        this.matlabBuilder.setTestRunTypeList(runOption);
+        project.getBuildersList().add(this.matlabBuilder);
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        jenkins.assertLogContains("-batch", build);
+        jenkins.assertLogContains("true,true,true", build);
+    }
+    
+    /*
+     * Test To verify if UI  throws an error when MATLAB root is empty.
+     * 
+     */
+
+    @Test
+    public void verifyEmptyMatlabRootError() throws Exception {
+        project.getBuildersList().add(this.matlabBuilder);
+        HtmlPage page = jenkins.createWebClient().goTo("job/test0/configure");
+        WebAssert.assertTextPresent(page, TestMessage.getValue("Builder.matlab.root.empty.error"));
+    }
+    
+    /*
+     * Test To verify UI does not throw any error when valid MATLAB root entered
+     * 
+     */
+
+    @Test
+    public void verifyInvalidValidMatlabRoot() throws Exception {
+        project.getBuildersList().add(this.matlabBuilder);
+        this.matlabBuilder.setLocalMatlab("/fake/matlab/path");
+        HtmlPage page = jenkins.createWebClient().goTo("job/test0/configure");
+        WebAssert.assertTextPresent(page, TestMessage.getValue("Builder.invalid.matlab.root.error"));
+    }
+    
+    /*
+     * Test To verify UI does not throw any error when valid MATLAB root entered
+     * 
+     */
+
+    @Test
+    public void verifyValidMatlabRoot() throws Exception {
+        project.getBuildersList().add(this.matlabBuilder);
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2018b"));
+        HtmlPage page = jenkins.createWebClient().goTo("job/test0/configure");
+        WebAssert.assertTextNotPresent(page, TestMessage.getValue("Builder.invalid.matlab.root.error"));
+    }
+    
+    /*
+     * Test To verify UI displays Cobertura Warning message when unsupported MATLAB version used.
+     * 
+     */
+
+    @Test
+    public void verifyCoberturaWarning() throws Exception {
+        project.getBuildersList().add(this.matlabBuilder);
+        this.matlabBuilder.setLocalMatlab(getMatlabroot("R2017a"));
+        HtmlPage page = jenkins.createWebClient().goTo("job/test0/configure");
+        HtmlCheckBoxInput coberturaChkBx = page.getElementByName("taCoberturaChkBx");
+        coberturaChkBx.setChecked(true);
+        Thread.sleep(2000);
+        WebAssert.assertTextPresent(page, TestMessage.getValue("Builder.matlab.cobertura.support.warning"));
+    }
+    
+    /*
+     * Test To verify UI displays Cobertura Error message when invalid MATLAB root entered.
+     * 
+     */
+
+    @Test
+    public void verifyCoberturaError() throws Exception {
```

Comment:
Should this be testing that Builder.matlab.cobertura.support.warning is there?  Otherwise it seems like the invalid matlab root test might cover this already.

Reply:
Correct , this Error is already verified however both these scenario uses MatlabReleaseInfo to check the MATLAb version and both of them they throw version not found exception. hence in the exception case our code coverage was not hitting the exception block. to cover the same this test is added.